### PR TITLE
feat: add gc jobs

### PR DESCRIPTION
**Pull request by rast1025**
We need garbage collector for completed and failed jobs

### CODE REVIEW

File: pkg/cmd/gc/jobs/jobs.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+				ageText := strings.TrimSuffix(age.Round(time.Minute).String(), "0s")
+				log.Logger().Infof("Deleted job %s in namespace %s as its age is: %s", job.Name, ns, ageText)
+			}
+		}
+	}
+	return errorutil.CombineErrors(errors...)
+}
+
+// MatchesJob returns true if this job can be garbage collected
+func (o *Options) MatchesJob(job *v1.Job) (bool, time.Duration) {
```
Comment (ankitm123) [inline pkg/cmd/gc/jobs/jobs.go:105]: Any reason to make this an exported function? Are you planning to use it outside this package?

Comment (rast1025) [reply pkg/cmd/gc/jobs/jobs.go]: No, not really

**Comment by ankitm123** [comment]
I dont have access to the logs, but u @rast1025 please run `make lint` and `make test` locally, and see if there are any issues ...

File: pkg/cmd/gc/jobs/jobs.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+	deleteOptions := metav1.DeleteOptions{}
+	errors := []error{}
+	for k := range jobList.Items {
+		job := jobList.Items[k]
+		matches, age := o.matchesJob(&job)
+		if matches {
+			err := jobInterface.Delete(ctx, job.Name, deleteOptions)
+			if err != nil {
+				log.Logger().Warnf("Failed to delete job %s in namespace %s: %s", job.Name, ns, err)
```
Comment (ankitm123) [inline pkg/cmd/gc/jobs/jobs.go:93]: NIT: if using %s, may be do err.Error()?

**Comment by ankitm123** [comment]
Also, it would be good to have a `dry-run` flag like other args.

**Comment by rast1025** [comment]
> I dont have access to the logs, but @rast1025 please run `make lint` and `make test` locally, and see if there are any issues ...

There are no issues

File: pkg/cmd/gc/jobs/jobs.go (hunk 2 above)
Comment (rast1025) [reply pkg/cmd/gc/jobs/jobs.go]: Done

**Pull request merged**